### PR TITLE
Maintenance: Don't use NumPys re-exported Python built-ins.

### PR DESCRIPTION
**Pull request by MSeifert04**
As discussed in #6603 (and https://github.com/numpy/numpy/pull/6103) it's
unnecessary and misleading to use the Python types from the NumPy namespace.

With this PR this uses the default NumPy dtype corresponding to these
Python types.

### CODE REVIEW

**Comment by pllim** [comment]
I'll let @bsipocz set the milestone.

**Comment by mhvk** [comment]
Yes, definitely good. Though note that `np.bool_`, etc., are not really `dtype` instances; hence, given that everything has to be interpreted anyway, it would be fine to use the python `bool` too But definitely *not* the imported ones, so this is a good improvement regardless.

**Comment by bsipocz** [comment]
relevant docs failure:

`/home/circleci/project/docs/table/construct_table.rst:753:Unknown target name: "np.bool".`

**Comment by MSeifert04** [comment]
@mhvk Oh, really? I was under the impression that these NumPy types represent the dtypes. The documentation [here](https://docs.scipy.org/doc/numpy/user/basics.types.html#data-types) also says that:

>  Once you have imported NumPy using [...] the dtypes are available as np.bool_, np.float32, etc.

**Comment by MSeifert04** [comment]
> relevant docs failure:

it's never as easy as it seems 😕

File: docs/table/construct_table.rst [hunk 1]
```diff
@@ -750,7 +750,7 @@ The column data values, shape, and data type are specified in one of two ways:
   Examples include:
 
   - Python non-string type (float, int, bool)
-  - Numpy non-string type (e.g. np.float32, np.int64, np.bool)
+  - Numpy non-string type (e.g. np.float32, np.int64)
```
Comment (MSeifert04) [inline docs/table/construct_table.rst]: I wasn't sure if `np.bool\_` would fix the doc problem, so I've removed it. Not sure if that's fine, but plain `bool` is mentioned in the line above.

**Comment by mhvk** [comment]
```
In [2]: isinstance(np.bool_, np.dtype)
Out[2]: False

In [3]: isinstance(np.dtype('?'), np.dtype)
Out[3]: True

In [4]: np.dtype('?') == np.bool_
Out[4]: True

In [5]: np.dtype('?') == bool
Out[5]: True

In [6]: np.bool_ == bool
Out[6]: False
```
But it doesn't matter much, really: I tried a few timings and they are all in a wash (can only be sped up by pre-defining actual dtypes, but even that doesn't help much).

**Comment by MSeifert04** [comment]
Always nice if equality isn't transitive 😕 😄

**Comment by bsipocz** [comment]
Thanks @MSeifert04!

**Pull request merged**